### DISCUSSION TITLE
How do i check if the vpn server is online?

### DISCUSSION THREAD

**Discussion by uacaman** [vpnhood/VpnHood#107] [Q&A]
Hello, 

I just started to mess around with VpnHood and was able to create a server and android app and connect. 

I would like to add to check if the sever is online and maybe check the latency before connecting. Is such information available ?

**Comment by trudyhood** [comment]
Hi,
We deliberately didn't add this feature to prevent anyone without a valid token from detecting VpnHoodServer. But now I think it can be a good idea to add an optional config to the server so let you query the server status.

**Comment by uacaman** [comment]
Right now i am opening a a connection to the  server and recording the delay, while this work as a ping of sorts and for each "ping" it is outputting the error "Received an unexpected EOF or 0 bytes from the transport stream". While it is not such a big deal i believe it might make others errors hard to find. 

I was thinking in adding a rest webapi inside the server, but that would require bind to another port and that might not be available. 

I am still getting acquainted to the VpnHood  code, but i can try to implement and do a pr. Should i try to add server status message before or after the tcpAuthenticateAsServerAsync ?

**Reply by trudyhood** [reply]
tcpAuthenticateAsServerAsync  is required to establish HTTPS! if you do anything before it, your connection is not an ordinary HTTPS.
You should do your job after that. Also, you can define your custom message.

**Reply by uacaman** [reply]
I was able to check if the server is online by connecting/disconnecting, the HelloRequest already has info about the server. I dont't really know if connecting is useful, once the vpn is connected when i try to check the status of the others servers, it will use the the current vpn connection. 

I guess i need to find some way to prevent my android app from using the vpn.

**Reply by trudyhood** [reply]
I think we can exclude the VpnHood app (itself) from using VPN, but it won't work on Windows Platform. 
* Maybe the better option is to get the list of other servers and exclude them
* Each Vpn profile can have many Server Addresses, we can exclude them from using the established VPN, but I am not sure that you want to add all your server's IP addresses into a single VpnHood profile.

**Comment by uacaman** [comment]
i am trying to figure out the profile part on android but i am unable to find it. Can you point me to where to look?

**Reply by trudyhood** [reply]
Sorry, What do you mean by Profile part? What are you looking for?

**Comment by uacaman** [comment]
O thought there it was possible to create profiles for connections. What i did was try to connect to a list a servers in turn, if the connections are successful in 30 seconds the server is online, otherwise the server is down.
